tensorflow/core/framework/dataset_options.proto

```diff
@@ -144,6 +144,12 @@ message OptimizationOptions {
   oneof optional_filter_parallelization {
     bool filter_parallelization = 18;
   }
+
+  // Whether to use gpu compatible memory prior to the 
+  // prefetch_to_device call
+  oneof optional_enable_gpu_compatible_memory {
+    bool enable_gpu_compatible_memory = 19;
```

Comment:
I think today we can already overlapping CPU->GPU transfer with GPU computation after changhuilin@'s change, right? So is the goal here to save one memcpy on CPU?

Reply:
Yes, and in addition it is a step towards using the pinned allocator for outputs that will get copied to the device, so it is more robust and more efficient.